lib/main_common.pm

```diff
@@ -167,6 +167,20 @@ sub setup_env {
         # Does _not_ work on aarch64
         set_var('UEFI_PFLASH', 1);
     }
+    # By default format DASD devices before installation
+    if (check_var('BACKEND', 's390x')) {
+        # Format DASD before the installation by default
+        # Skip format dasd before origin system installation by autoyast in 'Upgrade on zVM'
+        # due to channel not activation issue. Need further investigation on it.
+        my $format_dasd;
+        if (get_var('S390_DISK') || get_var('UPGRADE') || get_var('UPGRADE_ON_ZVM')) {
+            $format_dasd = 'never';
+        }
+        else {
+            $format_dasd = 'pre_install';
+        }
+        set_var('FORMAT_DASD', get_var('FORMAT_DASD', $format_dasd));
```

Comment:
Ternary operator?

Reply:
why?  it's longer and this is more clear imho

Reply:
I used it initially, but if we cannot simplify the condition, it's too long to be easily readable.

Reply:
what about like this:
my $no_format_dasd = get_var('S390_DISK') || get_var('UPGRADE') || get_var('UPGRADE_ON_ZVM'));
set_var('FORMAT_DASD', $no_format_dasd ? 'never' : 'pre_install') if !get_var('FORMAT_DASD');